components/Header.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  showFavoritesBtnForRoutes,
+  showSearchBarForRoutes,
+  showViewAllBtnForRoutes,
+} from "@/constants/const";
+import { useAuth } from "@/context/AuthContext";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+import Image from "next/image";
+import { logOut } from "@/store/firebaseHelpers";
+
+const SearchBar = ({
+  setGlobalQuery,
+}: {
+  setGlobalQuery: (query: string) => void;
+}) => {
+  const [searchQuery, setSearchQuery] = useState("");
```

Comment:
Instead of state - let's make it URL driven 
`http://localhost:3000/list?q=charizard` 
Something like that
So it makes it easier to deeplink
Also use `getServerSideProps` to fetch data on server and pre-render some cards